core/arch/arm/plat-stm32mp1/shared_resources.c

```diff
@@ -388,7 +388,8 @@ bool stm32mp_periph_is_non_secure(enum stm32mp_shres id)
 {
 	lock_registering();
 
-	return shres_state[id] == SHRES_NON_SECURE;
+	return shres_state[id] == SHRES_NON_SECURE ||
+	       shres_state[id] == SHRES_UNREGISTERED;
```

Comment:
Then the name is misleading. How about:
```C
bool stm32mp_periph_is_non_secure_or_unregistered()
{
	return (stm32mp_periph_is_non_secure(id) ||
		stm32mp_periph_is_unregistered(id));
}
```

Reply:
Sounds like what you want is `!stm32mp_periph_is_secure(id)` rather than `stm32mp_periph_is_non_secure(id)`, then? Anyway it's your call.

Reply:
You're right! I'll update.

Reply:
Not exactly what I meant. I thought you would use `!stm32mp_periph_is_secure(id)` at the call sites, and remove `stm32mp_periph_is_non_secure()` altogether. It hurts my eyes to see `is_non_secure()` check anything else that `== SHRES_NON_SECURE`.

Reply:
Ok, I'll propose something. In the end, I would like it to be clear when one reads the code.